### PR TITLE
fix: Allow Hastus import to work when a block has as-directeds and regular trips

### PR DESCRIPTION
**Pull request by joshlarson**
Asana Ticket: https://app.asana.com/0/1205718271156548/1208002136498399/f

### CODE REVIEW

**Comment by joshlarson** [comment]
The [error](https://github.com/mbta/skate/pull/2736#issuecomment-2276106543) mentioned above is resolved by marking `%AsDirected{}`'s as [not being revenue trips](https://github.com/mbta/skate/pull/2736/files#diff-6efd3069f4e6ec418138f852efb794bdceb49ef5b27e06785e9f0221c443e706R180) for the purpose of finding the next revenue trip for a block.

Before merging this, we should confirm that we're comfortable with changing the behavior of `Trip.revenue_trip?/1`.

File: assets/src/components/propertiesPanel/minischedule.tsx [hunk 1]
```diff
@@ -839,7 +837,10 @@ const isBreak = (activity: Piece | Break): activity is Break =>
   Object.prototype.hasOwnProperty.call(activity, "breakType")
 
 const isTrip = (trip: Trip | AsDirected): trip is Trip =>
-  Object.prototype.hasOwnProperty.call(trip, "id")
+  !Object.prototype.hasOwnProperty.call(trip, "kind")
```
Comment (firestack) [inline assets/src/components/propertiesPanel/minischedule.tsx:840]:
```suggestion
  !(trip && "kind" in trip)
```

File: assets/src/models/minischeduleData.ts [hunk 2]
```diff
@@ -122,10 +123,12 @@ const tripFromData = (tripData: TripData): Trip => ({
...
   kind: asDirectedData.kind,
   startTime: asDirectedData.start_time,
   endTime: asDirectedData.end_time,
 })
 
-const isTripData = (data: TripData | AsDirectedData): data is TripData =>
-  Object.prototype.hasOwnProperty.call(data, "id")
+const isAsDirectedData = (
+  data: TripData | AsDirectedData
+): data is AsDirectedData => Object.prototype.hasOwnProperty.call(data, "kind")
```
Comment (firestack) [inline assets/src/models/minischeduleData.ts:134]:
```suggestion
): data is AsDirectedData => !Object.hasOwn(trip, "kind")
```

**Review by firestack — APPROVED** [review]
@hannahpurcell I tried to pull this apart to no avail, so I went ahead and described each of the commits but I think we can merge this as is and I've found we can't really avoid the :id value or the fix to `AsDirectedRow`

File: test/schedule/block_test.exs [hunk 3]
```diff
@@ -80,6 +80,31 @@ defmodule Schedule.BlockTest do
            pieces: [@piece]
          )
 
+  @as_directed %AsDirected{
+    id: "ad1",
+    kind: :wap,
```
Comment (hannahpurcell) [inline test/schedule/block_test.exs:85]: 😂 um

```suggestion
    kind: :wad,
```
Interesting that this didn't lead to failed tests?

Comment (hannahpurcell) [reply test/schedule/block_test.exs:85]: After looking at it `next_revenue_trip`, I'm not concerned there isn't a failed test, as that function wasn't responsible for checking it. But why didn't elixir's typing catch this??

Comment (firestack) [reply test/schedule/block_test.exs]: oh wow thanks for catching this @hannahpurcell !

**Pull request merged**